lib/utils/mailup.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * Implements MailUp REST API
+ */
+
+import * as request from "request";
+import * as winston from "winston";
+
+import { Either, isLeft, left, right } from "fp-ts/lib/Either";
+
+// request timeout in milliseconds
+const TIMEOUT = 10000;
+
+type HttpMethod = "GET" | "POST";
+
+export interface ISmtpAuthInfo {
+  readonly Username: string;
+  readonly Secret: string;
+}
+
+export interface IApiResponse {
+  readonly Status: string;
+  readonly Code: string;
+  readonly Message: string;
+}
+
+function callMailUpApi<T extends IApiResponse>(
+  method: HttpMethod,
+  url: string,
+  creds: ISmtpAuthInfo,
+  payload: {}
+): Promise<Either<Error, T>> {
+  return new Promise(resolve => {
+    const options = {
+      uri: url,
+      method,
+      json: { ...payload, User: creds },
+      timeout: TIMEOUT
+    };
+    request(options, async (err, res, body) => {
```

Comment:
there's a Promise-enabled version of [request](https://www.npmjs.com/package/request#promises--asyncawait) (probably `request-promise-native` is what we need in this context)

Reply:
I was thinking to replace request with superagent as it could be useful to write integration tests as well https://github.com/visionmedia/supertest